src/portfolio-new/index.js

```diff
@@ -0,0 +1,380 @@
+// @flow
+
+/**
+ * GOALS
+ * - be non breaking changes for portfolio/index.js
+ * - fix bugs like countervalues portfolio refresh issue
+ * - be ready for "all time" range support
+ * - more performant
+ *
+ * MILESTONES
+ * - poc: implement the new methods
+ *   - test: update the existing tests with new impl
+ * - cli: port the cli
+ * - react: implement a react.js
+ * - perf: improve the impl to be performant
+ * - finish: connect to LLD / LLM
```

Comment:
could you recap this back into the PR description and try to note which of the GOALS and MILESTONES we set has been accomplished so far? thanks